.github/workflows/cd.yaml

```diff
@@ -26,10 +26,16 @@ jobs:
           TAG_NO_PREFIX=$(echo "$TAG" | sed 's/^refs\/tags\///')
           # Extract the path of the module to publish from the tag
           PACKAGE_PATH=$(echo "$TAG_NO_PREFIX" | rev | cut -d'/' -f2- | rev)          
-          # Save the path to the module for use in the build/publish step to only update the module associated with this tag
+          # Extract version to publish from tag
+          VERSION="${TAG_NO_PREFIX##*/}"
+          echo $VERSION
+          # Save the path to the module and the version for use in the build/publish step to only update the module associated with this tag
           echo "PACKAGE_PATH=$PACKAGE_PATH" >> $GITHUB_ENV
+          echo "VERSION=$VERSION" >> $GITHUB_ENV
       - name: Build & Publish to PyPi
         run: |
           pip install poetry
-          cd $PACKAGE_PATH         
+          cd $PACKAGE_PATH  
+          # replace version                 
+          sed -i "s/^version =.*/version = \"${VERSION}\"/" pyproject.toml 
```

Comment:
Yes we could do that.
But I just noticed that this might be a problem. If we don't write the version back (for the core) the core-package version in the requirements of the services and the version defined in the repo differs.
This will only be a problem for a local installation, but this still is an issue.
So we probably need to write the version back to the main branch. At least for the core.
Should this be done automatically (is this even possible with a protected main-branch?) or should I create a Pr in the pipeline that still needs manual approval?

Reply:
It should be possible if the pipeline has access to write to main, but it could also just open a PR. I just think that if it is a manual process it's not very scalable.

We could also consider the other way around, instead of **creating a tag triggering a release**, we could have a **PR editing the version in the `pyproject.toml` files triggering the creation of a new git tag** automatically

Reply:
That's a good idea. I like that